src/main/java/com/kmush12/CurrencyConverter/exchange/ExchangeRateProvider.java

```diff
@@ -11,8 +12,8 @@ public ExchangeRateProvider(ExternalExchangeRateClient externalExchangeRateClien
         this.externalExchangeRateClient = externalExchangeRateClient;
     }
 
-    public ExchangeRate getExchangeRate(String originCurrency, String destinationCurrency) {
-        return new ExchangeRate(externalExchangeRateClient.getExternalRate(originCurrency, destinationCurrency).base(), externalExchangeRateClient.getExternalRate(originCurrency, destinationCurrency).rates());
+    public ExchangeRate getExchangeRate(String originCurrency) {
+        return new ExchangeRate(externalExchangeRateClient.getExternalRate(originCurrency).base(), externalExchangeRateClient.getExternalRate(originCurrency).rates());
```

Comment:
Tutaj przez wywołanie `externalExchangeRateClient.getExternalRate` robisz dwa zapytania po http do zewnętrznego API. Zrób jedno zapytanie, przypisz wynik do zmiennej i z niej weź dopiero `.base()` i `.rates()`.